recipe/meta.yaml

```diff
@@ -140,11 +142,12 @@ outputs:
       requires:
         - {{ compiler("c") }}
         - {{ compiler("cxx") }}
-        - gxx
-        - gcc
+        - gxx              # [linux]
+        - gcc              # [linux]
+        - c-compiler       # [win]
```

Comment:
Is this needed given we have `{{ compiler("c") }}` above?

Reply:
I get the following error without it:
```
%PREFIX%\Library\include\crt\math_functions.h(11301): error: namespace "std" has no member "enable_if_t"

%PREFIX%\Library\include\crt\math_functions.h(11301): error: expected a "," or ">"

%PREFIX%\Library\include\crt\math_functions.h(11301): error: this declaration has no storage class or type specifier

%PREFIX%\Library\include\crt\math_functions.h(11301): internal error: assertion failed at: "error.c", line 1446 in form_template_arg_info
```

That said, I haven't figured out why `{{ compiler("c") }}` isn't sufficient here.

Reply:
It would be good to understand why that is. It may indicate a packaging issue that we need to address since recipes using `{{ compiler("cuda") }}` will just add `{{ compiler("c") }}` (and maybe `{{ compiler("cxx") }}`), but not the `c-compiler` (or `cxx-compiler`) package(s)

Also would add `c-compiler` on Windows is just [a wrapper around `{{ compiler("c") }}`]( https://github.com/conda-forge/compilers-feedstock/blob/95a25d8e5d853ebcacce1ad223f3196139d2497a/recipe/meta.yaml#L35-L43 ). It doesn't pull in anything additional. So it is a bit confusing why that would fix the issue

Reply:
Maybe try this? Though it is already pulled in by `{{ compiler("c") }}`

```suggestion
        - vc               # [win]
```

Reply:
No luck with this change. However looking closer at the packages being pulled into the test environment I do see the successful ones have `vs2019_win-64` and the failing has `vs2017_win-64`. So `c-compiler` is somehow bringing the newer version with it.

Reply:
It is odd that 2017 is being used. As noted below, `{{ compiler("c") }}` &  `{{ compiler("cxx") }}` should be configured to use 2019

Reply:
Maybe the `run_constrained` section above needs an update

Reply:
If you were building locally, you probably didn't use the corresponding `.ci_support/` file.

Reply:
It's a good point. Would make sure to run

```batch
conda.exe mambabuild recipe -m .ci_support\win_64_.yaml
```